src/variant.rs

```diff
@@ -144,10 +145,10 @@ impl Variant {
     }
 
     /// Constructs a new serialised-mode GVariant instance.
-    pub fn new_from_bytes<T: StaticVariantType>(bytes: &Bytes) -> Self {
+    pub fn new_from_bytes<T: DynamicVariantType>(bytes: &Bytes) -> Self {
         unsafe {
             from_glib_none(glib_sys::g_variant_new_from_bytes(
-                T::static_variant_type().as_ptr() as *const _,
+                T::variant_type().as_ptr() as *const _,
```

Comment:
This would now everywhere cause needless allocations for simple types. Maybe that can be solved with some `Cow`? :)

Reply:
not sure if that's possible w/o breaking the API but it's also that current API with lots of types is very confusing to me.